src/url_aggregator.cpp

```diff
@@ -1685,8 +1688,8 @@ inline void url_aggregator::consume_prepared_path(std::string_view input) {
               ? path_buffer_tmp
               : path_view;
       if (unicode::is_double_dot_path_segment(path_buffer)) {
-        helpers::shorten_path(path, type);
-        if (location == std::string_view::npos) {
+        if ((helpers::shorten_path(path, type) || special) &&
```

Comment:
order can be changed here as well

Reply:
Given...

```
((helpers::shorten_path(path, type) || special)
```

The compiler must evaluate `helpers::shorten_path(path, type)`. If it is true, it does not need to check `special`. But `special` is only checked if `helpers::shorten_path(path, type)` is false. This means that `helpers::shorten_path(path, type)` is almost computed, which is what we want.